pkg/topology/arangodb/arangodb-peer-change.go

```diff
@@ -0,0 +1,20 @@
+package arangodb
+
+func (a *arangoDB) peerChangeHandler(obj *message.PeerStateChange) {
+	glog.Infof("><SB> peer change handler - update db with peer change message data")
+
+	// creates a new BGP peer document in Arango's BGPPeer vertex collection
+	bgp_peer_document := &arangoDB.BGPPeer{
+		RouterID:	message.LocalBGPID
+		LocalIP:	message.LocalIP
+		LocalASN:	message.LocalASN
+		PeerIP:		message.RemoteIP
+		PeerBGPID:	message.RemoteBGPID
+		PeerASN:	message.RemoteASN
+	}
+	if err := a.Upsert(bgp_peer_document); err != nil {
```

Comment:
I do not see a direct call for Upsert. Did you try to compile your code? I suspect you had a helper function in your old topology which was used to do Upsert..  Please send me a line in the old topology where you used it. I will try to figure out.
Another thing is you need to check for the collection, open collection and then store it in it.

Reply:
ok, i think it is in one of these: